Cython/Utility/ObjectHandling.c

```diff
@@ -2000,15 +2020,51 @@ static int __Pyx_TryUnpackUnboundCMethod(__Pyx_CachedCFunction* target) {
     target->method = method;
 #if CYTHON_COMPILING_IN_CPYTHON
     #if PY_MAJOR_VERSION >= 3
-    // method descriptor type isn't exported in Py2.x, cannot easily check the type there
     if (likely(__Pyx_TypeCheck(method, &PyMethodDescr_Type)))
+    #else
+    // method descriptor type isn't exported in Py2.x, cannot easily check the type there.
+    // Therefore, reverse the check to the most likely alternative
+    // (which is returned for class methods)
+    if (likely(!PyCFunction_Check(method)))
```

Comment:
There is some code that already handles this case in the `ClassMethod` section of `CythonFunction.c`. Could be extracted into a reusable `__Pyx_PyMethodDescr_Check()` function (which PyPy supports already).

Reply:
IMHO, this fix deserves it's own pr (#5225), it's scope is different from what this pr tries to solve